src/components/application_manager/test/commands/mobile/add_command_request_test.cc

```diff
@@ -229,6 +249,97 @@ TEST_F(AddCommandRequestTest, OnEvent_VR_UNSUPPORTED_RESOURCE) {
   }
 }
 
+TEST_F(AddCommandRequestTest, OnEvent_UI_AddCommand_UNSUPPORTED_RESOURCE) {
+  MessageSharedPtr msg_vr = CreateFullParamsUISO();
+  (*msg_vr)[am::strings::msg_params][am::strings::vr_commands][0] =
+      "vr_command";
+
+  utils::SharedPtr<AddCommandRequest> req_vr =
+      CreateCommand<AddCommandRequest>(msg_vr);
+
+  MockAppPtr mock_app = CreateMockApp();
+  ON_CALL(app_mngr_, application(kConnectionKey))
+      .WillByDefault(Return(mock_app));
+  ON_CALL(*mock_app, app_id()).WillByDefault(Return(kAppId));
+  ON_CALL(*mock_app, FindSubMenu(_)).WillByDefault(Return(&(*msg_vr)));
+  MockHmiInterfaces hmi_interfaces;
+  ON_CALL(app_mngr_, hmi_interfaces()).WillByDefault(ReturnRef(hmi_interfaces));
+  ON_CALL(hmi_interfaces, GetInterfaceFromFunction(_))
+      .WillByDefault(
+          Return(am::HmiInterfaces::HMI_INTERFACE_BasicCommunication));
+  ON_CALL(hmi_interfaces, GetInterfaceState(_))
+      .WillByDefault(Return(am::HmiInterfaces::STATE_AVAILABLE));
+
+  MessageSharedPtr msg = CreateMessage(smart_objects::SmartType_Map);
+  (*msg)[am::strings::params][am::hmi_response::code] =
+      hmi_apis::Common_Result::UNSUPPORTED_RESOURCE;
+  (*msg)[am::strings::msg_params][am::strings::cmd_id] = kCommandId;
+
+  Event event(hmi_apis::FunctionID::UI_AddCommand);
+  event.set_smart_object(*msg);
+
+  smart_objects::SmartObject* ptr = NULL;
+  ON_CALL(*mock_app, FindCommand(kCmdId)).WillByDefault(Return(ptr));
+  EXPECT_EQ(NULL, ptr);
+
+  MockMessageHelper* mock_message_helper =
+      MockMessageHelper::message_helper_mock();
+  ON_CALL(*mock_message_helper, HMIToMobileResult(_))
+      .WillByDefault(Return(mobile_apis::Result::SUCCESS));
+
+  ON_CALL(*mock_message_helper, VerifyImage(_, _, _))
+      .WillByDefault(Return(mobile_apis::Result::SUCCESS));
+  EXPECT_CALL(*mock_app,
+              AddCommand(kCmdId, (*msg_vr)[am::strings::msg_params]));
+
+  am::CommandsMap commands_map;
+  ON_CALL(*mock_app, commands_map())
+      .WillByDefault(
+          Return(DataAccessor<am::CommandsMap>(commands_map, lock_)));
+
+  EXPECT_CALL(
+      app_mngr_,
+      ManageMobileCommand(_, am::commands::Command::CommandOrigin::ORIGIN_SDL))
+      .WillOnce(Return(true))
+      .WillOnce(Return(true));
+
+  EXPECT_CALL(*mock_app, UpdateHash());
+
+  MessageSharedPtr msg_ui(CreateMessage(smart_objects::SmartType_Map));
+  (*msg_ui)[am::strings::params][am::hmi_response::code] =
+      hmi_apis::Common_Result::SUCCESS;
+  (*msg_ui)[am::strings::params][am::strings::info] = "info";
+
+  Event event_vr(hmi_apis::FunctionID::VR_AddCommand);
+  event_vr.set_smart_object(*msg_vr);
+
+  req_vr->Run();
+  req_vr->on_event(event_vr);
+
+  MessageSharedPtr ui_command_result;
+  EXPECT_CALL(
+      app_mngr_,
+      ManageMobileCommand(_, am::commands::Command::CommandOrigin::ORIGIN_SDL))
+      .WillOnce(DoAll(SaveArg<0>(&ui_command_result), Return(true)));
```

Comment:
@okozlovlux, you can use `CatchMobileCommandResult()` method of `CommandRequestTest` class, in this kind of case